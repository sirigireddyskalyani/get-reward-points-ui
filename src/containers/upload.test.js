import React from 'react';
import { Provider } from 'react-redux';
import { screen, render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Upload from './upload';

const mockStore = configureStore([]);
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => (jest.fn())
}));

describe('Upload Container Snapshot', () => {
    let store;
    let container;
    let component;
    let testInstance;

    beforeEach(() => {
        store = mockStore({
            expenses: {
                data: []
            }
        });

        component = renderer.create(
            <Provider store={store}>
                <Upload />
            </Provider>
        );
        testInstance = component.root;
        container = component.container;
    });

    it('Upload container rendered successfully', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
});

describe('Upload Container Behavior Testing', () => {
    let store;

    beforeEach(() => {
        store = mockStore({
            expenses: {
                data: []
            }
        });
        jest.clearAllMocks();
        jest.spyOn(global, 'FileReader').mockImplementation(function () {
            this.onload = jest.fn();
            this.readAsText=jest.fn();
        });
    });

    it('on Upload submit button click', () => {
        const { container } = render(
            <Provider store={store}>
                <Upload />
            </Provider>
        );
        const uploadSubmit = screen.getAllByTestId('upload-submit-btn')[0];
        uploadSubmit.click();
        expect(1).toEqual(1);
    });

    it('on Upload Back button click', () => {
        const { container } = render(
            <Provider store={store}>
                <Upload />
            </Provider>
        );
        const uploadSubmit = screen.getAllByTestId('upload-back-btn')[0];
        uploadSubmit.click();
        expect(1).toEqual(1);
    });

    it('on Fileupload', async () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <Upload />
            </Provider>
        );
        const file = new File([{name: 'expense_data.json', lastModified: 1655699056007, webkitRelativePath: '', size: 12768}], "chucknorris.png", { type: "image/png" });
        let uploader = getByTestId("expenses-json-uploader");
        await fireEvent.change(uploader, {
            target: { files: [file] },
            currentTarget: { files: [file] },
        })
        const uploadSubmit = screen.getAllByTestId('upload-back-btn')[0];
        uploadSubmit.click();
        expect(1).toEqual(1);
    });
});