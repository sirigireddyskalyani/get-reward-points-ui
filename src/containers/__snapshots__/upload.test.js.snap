// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Container Snapshot Upload container rendered successfully 1`] = `
<div
  className="m-4 flex flex-col items-center justify-center"
>
  <div
    className="card w-full bg-base-100 shadow-xl"
  >
    <div
      className="card-body"
    >
      <div
        className="flex flex-row"
      >
        <button
          className="btn btn-sm m-2 w-16"
          data-testid="upload-back-btn"
          onClick={[Function]}
        >
          Back
        </button>
        <h2
          className="card-title"
        >
          Upload
        </h2>
      </div>
      <div
        className="flex flex-col mt-4"
      >
        <div
          className="w-full m-1 flex justify-end"
        >
          <a
            className="btn btn-link btn-sm"
            download={true}
            href="./expense_data.json"
          >
            Download Sample JSON
          </a>
        </div>
        <div
          className="w-full m-1"
        >
          <label
            className="form-label inline-block mb-2"
            htmlFor="formFile"
          >
            Upload Expenses JSON file
          </label>
          <input
            accept=".json"
            className="form-control block w-full px-3 py-1.5 text-base font-normal bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
            data-testid="expenses-json-uploader"
            id="formFile"
            onChange={[Function]}
            type="file"
          />
        </div>
        <div
          className="w-full m-1 flex justify-center"
        >
          <button
            className="btn btn-sm btn-success m-2 w-24"
            data-testid="upload-submit-btn"
            onClick={[Function]}
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
